middlewares/auth.js

```diff
@@ -0,0 +1,40 @@
+const passport = require("passport");
+require("dotenv").config();
+
+const auth = (requiredSubscription) => (req, res, next) => {
+    passport.authenticate("jwt", { session: false }, (err, user) => {
+
+        if (!req.get('Authorization')) {
+            return res.status(401).json({
+                status: 'error',
+                message: 'Unauthorized',
+                data: 'Unauthorized',
+            });
+        };
+
+        const token = req
+            .get('Authorization')
+            .replace('Bearer ', '');
+
+        if (!user || err || !token || token !== user.token) {
+            return res.status(401).json({
+                status: "error",
+                message: "Unauthorized",
+                data: "Unauthorized",
+            });
+        };
```

Comment:
Tu możesz to skrócić dodając "?" jako "jeśli istnieje" i wtedy będzie tylko jeden if
```
    const token = req.get('Authorization')?.replace('Bearer ', '');

    if (!user || err || !token || token !== user.token) {
      return res.status(401).json({
        status: "error",
        message: "Unauthorized",
        data: "Unauthorized",
      });
    }
```